x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/utils/MlIndexAndAliasTests.java

```diff
@@ -116,6 +120,34 @@ public void verifyNoMoreInteractionsWithMocks() {
         verifyNoMoreInteractions(indicesAdminClient, listener);
     }
 
+    public void testInstallIndexTemplateIfRequired_TemplateExist() {
+        ClusterState clusterState = createClusterState(Collections.emptyMap(),
+            Collections.singletonMap(InferenceIndexConstants.LATEST_INDEX_NAME,
+                createIndexTemplateMetaData(InferenceIndexConstants.LATEST_INDEX_NAME,
+                    Collections.singletonList(InferenceIndexConstants.LATEST_INDEX_NAME))));
+
+        IndexTemplateConfig inferenceTemplate = new IndexTemplateConfig(InferenceIndexConstants.LATEST_INDEX_NAME,
+            "not_a_real_file.json", Version.CURRENT.id, "xpack.ml.version",
+            Collections.singletonMap("xpack.ml.version.id", String.valueOf(Version.CURRENT.id)));
+
+        MlIndexAndAlias.installIndexTemplateIfRequired(clusterState, client, inferenceTemplate, listener);
+        verify(listener).onResponse(true);
+        verifyNoMoreInteractions(client);
+    }
+
+    public void testInstallIndexTemplateIfRequired() {
+        ClusterState clusterState = createClusterState(Collections.emptyMap());
+
+        IndexTemplateConfig inferenceTemplate = new IndexTemplateConfig(InferenceIndexConstants.LATEST_INDEX_NAME,
+            "/org/elasticsearch/xpack/core/ml/inference_index_template.json", Version.CURRENT.id, "xpack.ml.version",
+            Collections.singletonMap("xpack.ml.version.id", String.valueOf(Version.CURRENT.id)));
+
+        MlIndexAndAlias.installIndexTemplateIfRequired(clusterState, client, inferenceTemplate, listener);
+        InOrder inOrder = inOrder(indicesAdminClient, listener);
+        inOrder.verify(indicesAdminClient).putTemplate(any(), any());
+        inOrder.verify(listener).onResponse(true);
```

Comment:
Could add the following line here?
```
inOrder.verifyNoMoreInteractions();
```

Reply:
Good point. There is an `After` method which calls `verifyNoMoreInteractions` on the mocks but you couldn't have known that from the diff

Reply:
Actually, I should have known since I wrote the method ;) I just forgot about it :/